newm-server/src/main/kotlin/io/newm/server/features/marketplace/repo/MarketplaceRepositoryImpl.kt

```diff
@@ -141,6 +130,114 @@ internal class MarketplaceRepositoryImpl(
         }
     }
 
+    override suspend fun generateSaleAmount(request: SaleAmountRequest): SaleAmountResponse {
+        log.debug { "generateCreateSaleAmount: $request" }
+
+        val nativeAssets =
+            listOf(
+                nativeAsset {
+                    policy = request.bundlePolicyId
+                    name = request.bundleAssetName
+                    amount = computeAmount(request.bundleAmount, request.totalBundleQuantity)
+                }
+            )
+
+        val lovelace = configRepository.getLong(CONFIG_KEY_MARKETPLACE_SALE_LOVELACE)
```

Comment:
We need to add 1 Ada to this amount (1,000,000 lovelace). This way, when the wallet selects UTxOs, it will always have enough extra to send back some change. Otherwise, it might select a utxo with the exact lovelace we're asking for and then the tx to create the sale will fail due to minUtxoNotMet.